hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java

```diff
@@ -143,9 +144,11 @@ public BackgroundTaskResult call() throws Exception {
 
           String dbName = OM_DB_NAME + snapshotInfo.getCheckpointDirName();
 
+          String snapshotCheckpointDir = omMetadataDir + OM_KEY_PREFIX +
+              OM_SNAPSHOT_CHECKPOINT_DIR;
           RDBStore rdbStore = (RDBStore) OmMetadataManagerImpl
-              .loadDB(ozoneManager.getConfiguration(), new File(snapshotDir),
-                  dbName, true);
+              .loadDB(ozoneManager.getConfiguration(),
+                  new File(snapshotCheckpointDir), dbName, true);
           RocksDatabase db = rdbStore.getDb();
           db.deleteFilesNotMatchingPrefix(prefixPairs, filterFunction);
 
```

Comment:
Where are we serializing sst file filtering service with tarball creation ?

Reply:
I discussed that briefly [above](https://github.com/apache/ozone/pull/3980#issuecomment-1426468318).  Basically I added most of the code for the serialization and then pulled it out because the PR is getting to big.  I'll be ready to create a separate PR soon after this is merged.

Reply:
added ticket here: https://issues.apache.org/jira/browse/HDDS-8209